notifier/notifier.go

```diff
@@ -383,25 +391,25 @@ func (n *Notifier) Stop() {
 
 // Describe implements prometheus.Collector.
 func (n *Notifier) Describe(ch chan<- *prometheus.Desc) {
```

Comment:
It has nothing to do with optimization. Optimization means adding more complexity in favor of something else. This is not about complexity but about general design.
And it works best if applied consistently. So it is actually pretty complex to move afterwards "if someone asks". We have seen this in huge refactorings happening before when we had all flags at a global per-package level.

Saying this here as someone feeling the pain of global flags, metrics, and registries on an almost daily basis. It never works as soon as any 2 open source projects have to play well together.